tests/integration/test_kubernetes_worker_integration.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from urllib.request import urlretrieve
+
+import pytest
+
+
+CNI_AMD64_URL = "https://api.jujucharms.com/charmstore/v5/~containers/kubernetes-worker-743/resource/cni-amd64/747"  # noqa
+
+log = logging.getLogger(__name__)
+
+
+def _check_status_messages(ops_test):
+    """ Validate that the status messages are correct. """
+    expected_messages = {
+        "kubernetes-master": "Kubernetes master running.",
+        "kubernetes-worker": "Kubernetes worker running.",
+    }
+    for app, message in expected_messages.items():
+        for unit in ops_test.model.applications[app].units:
+            assert unit.workload_status_message == message
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    cni_amd64 = ops_test.tmp_path / "cni-amd64.tgz"
+    urlretrieve(CNI_AMD64_URL, cni_amd64)
```

Comment:
Not a fan of the hard-coded CNI resource. What if someone changes the [build script](https://github.com/charmed-kubernetes/charm-kubernetes-worker/blob/master/build-cni-resources.sh)? The test wouldn't cover those changes.

I would consider building the resource here, just like we build the charm.